src/main/java/appeng/parts/automation/PartAbstractFormationPlane.java

```diff
@@ -0,0 +1,264 @@
+
+package appeng.parts.automation;
+
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+
+import appeng.api.config.Actionable;
+import appeng.api.networking.security.IActionSource;
+import appeng.api.parts.IPart;
+import appeng.api.parts.IPartCollisionHelper;
+import appeng.api.parts.IPartHost;
+import appeng.api.storage.ICellContainer;
+import appeng.api.storage.IMEInventory;
+import appeng.api.storage.data.IAEStack;
+import appeng.api.storage.data.IItemList;
+import appeng.api.util.AECableType;
+import appeng.api.util.AEPartLocation;
+import appeng.api.util.IConfigManager;
+import appeng.helpers.IPriorityHost;
+
+
+public abstract class PartAbstractFormationPlane<T extends IAEStack<T>> extends PartUpgradeable implements ICellContainer, IPriorityHost, IMEInventory<T>
+{
+
+	private boolean wasActive = false;
+	private int priority = 0;
+	protected boolean blocked = false;
+
+	public PartAbstractFormationPlane( ItemStack is )
+	{
+		super( is );
+	}
+
+	protected abstract void updateHandler();
+
+	@Override
+	protected int getUpgradeSlots()
+	{
+		return 5;
+	}
+
+	@Override
+	public void upgradesChanged()
+	{
+		this.updateHandler();
+	}
+
+	@Override
+	public void updateSetting( final IConfigManager manager, final Enum settingName, final Enum newValue )
+	{
+		this.updateHandler();
+		this.getHost().markForSave();
+	}
+
+	public void stateChanged()
+	{
+		final boolean currentActive = this.getProxy().isActive();
+		if( this.wasActive != currentActive )
+		{
+			this.wasActive = currentActive;
+			this.updateHandler();
+			this.getHost().markForUpdate();
+		}
+	}
+
+	@Override
+	public void getBoxes( final IPartCollisionHelper bch )
+	{
+		int minX = 1;
+		int minY = 1;
+		int maxX = 15;
+		int maxY = 15;
+
+		final IPartHost host = this.getHost();
+		if( host != null )
+		{
+			final TileEntity te = host.getTile();
+
+			final BlockPos pos = te.getPos();
+
+			final EnumFacing e = bch.getWorldX();
+			final EnumFacing u = bch.getWorldY();
+
+			if( this.isTransitionPlane( te.getWorld().getTileEntity( pos.offset( e.getOpposite() ) ), this.getSide() ) )
+			{
+				minX = 0;
+			}
+
+			if( this.isTransitionPlane( te.getWorld().getTileEntity( pos.offset( e ) ), this.getSide() ) )
+			{
+				maxX = 16;
+			}
+
+			if( this.isTransitionPlane( te.getWorld().getTileEntity( pos.offset( u.getOpposite() ) ), this.getSide() ) )
+			{
+				minY = 0;
+			}
+
+			if( this.isTransitionPlane( te.getWorld().getTileEntity( pos.offset( u ) ), this.getSide() ) )
+			{
+				maxY = 16;
+			}
+		}
+
+		bch.addBox( 5, 5, 14, 11, 11, 15 );
+		bch.addBox( minX, minY, 15, maxX, maxY, 16 );
+	}
+
+	public PlaneConnections getConnections()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "getConnections" is 13 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)